recipes/itk/all/test_package/conanfile.py

```diff
@@ -1,17 +1,27 @@
-from conans import ConanFile, CMake, tools
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout, CMake
 import os
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
-    generators = "cmake", "cmake_find_package_multi"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
+    test_type = "explicit"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+        self.requires("hdf5/[*]") 
```

Comment:
Why do you need to include `hdf5` now? Please, try to keep the test package simple enough, by not using other extra dependencies, besides transitives dependencies from the `itk`

Reply:
Conan v2 doesn't provide hdf5 includes transitively - only libs when they are needed. Test is directly including hdf5. I didn't write this test - it was added here
https://github.com/conan-io/conan-center-index/commit/4c98033ee8ddb4ef6e1a062fd1bcc0992021dd4f#diff-434b1d10721bd060ddb97e45964989be03a412b0f360f58ae7d1c75ca6321850

Reply:
@Nekto89 it's possible to enable transitive libs and headers: https://docs.conan.io/2/reference/conanfile/methods/requirements.html#transitive-headers

We have some usage here in CCI, so few free to update it in conanfile.py with hdf5 requirement, so you can remove it from here in test package.

Reply:
I don't think that hdf5 should be in "INTERFACE" part. In ITK hdf5 headers don't leak to users as far as I can see from source code and are only used during build.
https://github.com/search?q=repo%3AInsightSoftwareConsortium%2FITK%20itk_H5Cpp.h&type=code

There is some mentioning of hdf5 problems in validate()
```
if self.options.shared and not self.options["hdf5"].shared:
            raise ConanInvalidConfiguration("When building a shared itk, hdf5 needs to be shared too (or not linked to by the consumer).\n"
                                            "This is because H5::DataSpace::ALL might get initialized twice, which will cause a H5::DataSpaceIException to be thrown).")
```

Maybe that's what was tested here at some point?

I can remove hdf5 completely from test package if no one sees any value in it (personally I don't).

Reply:
> I can remove hdf5 completely from test package if no one sees any value in it (personally I don't).

Yes, please.